### PR TITLE
Add a storage implementation for ThirdPartyResources.

### PR DESCRIPTION
**Pull request by brendandburns**
@lavalamp @JanetKuo

### CODE REVIEW

**Comment by brendandburns** [comment]
@krousey for CSI work

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by JanetKuo** [comment]
cc @caesarxuchao

File: pkg/expapi/v1/defaults.go [hunk 1]
```diff
@@ -16,5 +16,13 @@ limitations under the License.
 
 package v1
 
+import "k8s.io/kubernetes/pkg/api"
+
 func addDefaultingFuncs() {
+	api.Scheme.AddDefaultingFuncs(
+		func(obj *APIVersion) {
```
Comment (caesarxuchao) [inline pkg/expapi/v1/defaults.go]: This defaulting function is also added in api/v1/defaults.go. I think the duplication is unnecessary.

**Comment by caesarxuchao** [comment]
LGTM except a nit. Thank you.

File: pkg/expapi/v1/defaults.go (hunk 1 above)
Comment (brendandburns) [inline pkg/expapi/v1/defaults.go]: Removed.

**Comment by brendandburns** [comment]
Comment addressed, also addressed the change to `c.ExpDatabaseStorage` from the other PR.

Thanks
--brendan

**Comment by caesarxuchao** [comment]
LGTM. Thank you.

**Pull request merged**